File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/Combined/SearchComponent.java
package com.amakakeru.mangaworld.Entity.Combined;

import com.amakakeru.mangaworld.Entity.Author;
import com.amakakeru.mangaworld.Entity.Manga;
import com.amakakeru.mangaworld.Entity.Publisher;
import lombok.Data;

import java.util.List;

@Data
public class SearchComponent {
    List<Manga> mangas;
    List<Author> authors;
    List<Publisher> publishers;
}

File: Frontend/manga-world/src/Model/Combined/SearchComponent.ts
import Manga from "../Manga";
import Author from "../Author";
import Publisher from "../Publisher";

class SearchComponent {
  mangas: Manga[];
  authors: Author[];
  publishers: Publisher[];

  constructor(mangas: Manga[], authors: Author[], publishers: Publisher[]) {
    this.mangas = [];
    this.authors = [];
    this.publishers = [];
  }
}

export default SearchComponent;
